src/test/java/com/atlassian/db/replica/it/DualConnectionIT.java

```diff
@@ -79,6 +79,22 @@ public void shouldRunNextValOnMainDatabase() throws SQLException {
         }
     }
 
+    @Test
+    public void shouldChangeTransactionIsolationLevel() throws SQLException {
```

Comment:
Why should it change level, when?
Could we put an actual contract in the name, I can read from test impl that it's changing level - but why?

Reply:
Changing level is part of the connection's API. It should change the level when you call the API. Maybe more precise would be shluldNotFailWhenChangingTransactionIsolationLevel WDYT?

Reply:
It doesn't assert on the new isolation level. So indeed `shouldNotFailWhen...` or the assertion.